src/index.ts

```diff
@@ -264,77 +352,86 @@ Upload.prototype.makeRequest = function (reqOpts, callback) {
     reqOpts.qs.userProject = this.userProject;
   }
 
-  this.authClient.authorizeRequest(reqOpts, function (err, authorizedReqOpts) {
-    if (err) return callback(wrapError('Could not authenticate request', err));
+  this.authClient.authorizeRequest(reqOpts, (err: Error, authorizedReqOpts: r.Options) => {
+    if (err) {
+      err = wrapError('Could not authenticate request', err)
+      return callback(err, null!, null);
+    }
 
-    request(authorizedReqOpts, function (err, resp, body) {
-      if (err) return callback(err, resp);
+    request(authorizedReqOpts, (err, resp, body) => {
+      if (err) {
+        return callback(err, resp, body);
+      }
 
-      if (body && body.error) return callback(body.error, resp);
+      if (body && body.error) {
+        return callback(body.error, resp, body);
```

Comment:
Could we use a new test to make sure `body` is passed?

Reply:
I modified `should execute the callback with error & response` under `makeRequest` to make sure this is tested as well.